File: app/src/main/java/com/noahkim/rolltime/FirebaseHelper.java
package com.noahkim.rolltime;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.database.DatabaseReference;

public class FirebaseHelper {

    // Firebase instance variables

    private FirebaseAuth mFirebaseAuth;
    private FirebaseAuth.AuthStateListener mAuthStateListener;
    private DatabaseReference mDatabaseReference;
    private static FirebaseHelper INSTANCE;


    public static FirebaseHelper getFirebaseInstance(){
        if (INSTANCE == null) {
            INSTANCE = new FirebaseHelper();
        }
        return INSTANCE;
    }

    public FirebaseHelper(){
        mFirebaseAuth = FirebaseAuth.getInstance();
    }
}

File: app/src/main/java/com/noahkim/rolltime/FirebaseHelper.kt
package com.noahkim.rolltime

import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.DatabaseReference

class FirebaseHelper {

    // Firebase instance variables
    private val mFirebaseAuth: FirebaseAuth

    init {
        mFirebaseAuth = FirebaseAuth.getInstance()
    }
}
